pages/index.tsx

```diff
@@ -1,8 +1,93 @@
+import axios from 'axios';
+import Link from 'next/link';
 import Head from 'next/head';
+import styled from 'styled-components';
+import { useEffect, useState } from 'react';
 import type { NextPage } from 'next';
 import styles from '../styles/Home.module.css';
 
+interface PostProps {
+  num: number;
+  title: string;
+  shortUrl: string;
+  userPosted: string;
+  timeCreated: string;
+  numberOfComments: number;
+}
+
+const PostWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+`;
+
+const PostInfo = styled.div`
+  display: flex;
+  gap: 0.5rem;
+  font-size: 14px;
+  margin-top: 0.4rem;
+
+  p {
+    margin: 0;
+  }
+`;
+
+const PostStats = styled.div`
+  display: flex;
+  font-size: 13px;
+
+  p {
+    margin: 0.3rem;
+  }
+`;
+
 const Home: NextPage = () => {
+  const [posts, setPosts] = useState<any>([]);
+
+  useEffect(() => {
+    const getAllPosts = async () => {
+      try {
+        const response = await axios.get('/api/allPosts');
```

Comment:
I would advise using `fetch`. Not important, I'll leave it up to you. But axios uses deprecated APIs under the scene. We want to be modern :)